app/controllers/region_admin/donors_controller.rb

```diff
@@ -0,0 +1,36 @@
+class RegionAdmin::DonorsController < ApplicationController
+  before_filter :authenticate_volunteer!
+  before_filter :authorize_region_admin!
+
+  def index
+    @locations = regional_donors
+    @regions   = available_regions
+  end
+
+  private
+
+  def regional_donors
+    if current_volunteer.super_admin?
+      Location.active.donors
+    else
+      Location.active.donors.regional(available_regions.map(&:id))
+    end
+  end
+
+  def available_regions
+    if current_volunteer.super_admin?
+      Region.all
+    else
+      current_volunteer.assignments.collect{ |a| a.admin ? a.region : nil }.compact
```

Comment:
I got involved in cleaning these calls up but it just added more changes/LOC and didn't fit the purpose of this refactor.  I agree this can be great soon! Very bigly!